app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/firestore/FirestoreDatabaseService.kt

```diff
@@ -71,37 +95,146 @@ class FirestoreDatabaseService(
         }
     }
 
-    override suspend fun getAllPictures(category: Category): Set<CategorizedPicture> {
-        TODO("Not yet implemented")
+    @OptIn(ExperimentalStdlibApi::class)
+    override suspend fun getAllPictures(category: Category): Set<FirestoreCategorizedPicture> {
+        require(category is FirestoreCategory)
+        categories.document(category.id).get().addOnCompleteListener {
```

Comment:
errors aren't going to propagate up if you don't `await`. also it probably already carries an exception so you can get away with
```kotlin
try {
  categories.document(category.id).get().await()
} catch (e: FirestoreExceptionOrSomething) {
  throw IllegalArgumentException("Category with id ${category.id} is not present in the database"
}

Reply:
Sadly it does not already carry an exception as far as I know. I improved all the similar code using `await`.